### ISSUE TITLE
Emmet preferences config not working and get notice box!

### ISSUE THREAD

**Issue by ngoncom** [eight04/jn-npp-emmet#15]
Hi,

I type "mr" and using key trigger (Ctrl + `) many time, I get this:

![1](https://user-images.githubusercontent.com/8530544/38456297-68fb5916-3aad-11e8-80e4-829f9e409989.png)

I want update some Emmet preferences (https://docs.emmet.io/customization/preferences/) inside:
%appdata%\Notepad++\plugins\config\emmet.settings.json but it not work, ex:

```
{
	"preferences": {
		"css.valueSeparator": ":",
		"css.propertyEnd": "; "
	},
	"enableTabExpensionByFileType": true,
	"enableTabExpensionUnder": ["html", "css"]
}
```

Thank so much!

P/S: with long css file, it take some seconds for Emmet working!

**Comment by eight04** [comment]
> I type "mr" and using key trigger (Ctrl + `) many time, I get this:

I can't reproduce it with this example:
```css
body {
  mr|
}
```

> I want update some Emmet preferences

Try to create this file:
```
%appdata%\Notepad++\plugins\config\emmet.preferences.json 
```
With following content:
```json
{
  "css.valueSeparator": ":",
  "css.propertyEnd": "; "
}
```
This feature should be documented. See #16.

> with long css file, it take some seconds for Emmet working!

I think the problem is that Emmet is too slow ([we had already rewritten the HTML matcher](https://github.com/eight04/jn-npp-emmet/issues/2)). Could you post the steps to reproduce? I can't promise when to rewrite the CSS matcher though.

**Comment by ngoncom** [comment]
> I can't reproduce it with this example:

In my case, I open a .css file and type "mr" it will expand with "margin-right: ;".

The error (see image above) will appear if I type "Ctrl + E" twice again (or using Ctrl + E multi times), after error notice I can't  "mr" to expand text any more!

![2](https://user-images.githubusercontent.com/8530544/38457921-ea887582-3ac0-11e8-8c4c-8f3c1f362acd.png)

Notice that: sometime it work properly, I don't know why!

> %appdata%\Notepad++\plugins\config\emmet.preferences.json

After create "emmet.preferences.json", the preferences work well!

Thank so much!

P/S: I am using
- Windows 7 (64 bits)
- Notepad++ v7.5.6 (32 bits) (fresh install to test)
- jn-npp-plugin: jN_2.2.185.2_x86.zip
- jn-npp-emmet: v1.2.0.zip

**Comment by eight04** [comment]
> The error (see image above) will appear if I type "Ctrl + E" twice again (or using Ctrl + E multi times), after error notice I can't "mr" to expand text any more!

Which command would be executed when you type "Ctrl + E"? Also, uploading the CSS file which generates the error would help.

**Comment by ngoncom** [comment]
> Which command would be executed when you type "Ctrl + E"?

I am using default keymap: Expand Abbreviation (Ctrl +E), and I check it not conflict with any another in Notepad++ Shortcut Mapper

> Also, uploading the CSS file which generates the error would help.

Here it is: http://www.mediafire.com/file/epwtiaw5qcli2ew/style.css

Thank for your support!

**Comment by eight04** [comment]
OK, I can reproduce the bug with following example:
```css
body {
  margin-top|
}
```
After expanding the expression (no matter which shortcut is used), Emmet stopped working.

**Comment by eight04** [comment]
Could you try https://github.com/eight04/jn-npp-emmet/pull/17?

**Comment by ngoncom** [comment]
After add some code as you mention, it work more properly!

In the rare case, if I using unicode and click Ctrl+E it will notice, ex: typing "ôa" and Ctrl+E. It will be perfect if you can fix this, but I think it is not important!

Another problem is much more keymap have been assigned, so that it conflict with Notepad++ core function, ex: Ctrl+D (duplicate line in Notepad++), I think you should set it is "null" as default value in /config/emmet.keymap.json. Because if I click "Ctrl+/" (toggle comment) in .css file (not in .html file), it will open notice pop-up.

Thank so much for your hard working!

**Comment by eight04** [comment]
Here is the issue for "out of stack space" bug: https://github.com/eight04/emmet/issues/1.

> Another problem is much more keymap have been assigned, so that it conflict with Notepad++ core function, ex: Ctrl+D (duplicate line in Notepad++)

How about Ctrl+Alt+D? It looks pretty consistent with "Balance inward" (Ctrl+Shift+D).

**Comment by ngoncom** [comment]
> How about Ctrl+Alt+D? It looks pretty consistent with "Balance inward" (Ctrl+Shift+D).

Thank for your suggestion and your time, that is useful keymap!

**Comment by eight04** [comment]
https://github.com/eight04/jn-npp-emmet/issues/18

**Comment by eight04** [comment]
Released v1.2.2.
